lib/mongoProvider.js

```diff
@@ -103,7 +103,7 @@ MongoCollection.prototype.find = function (query, req) {
     return self.db().then(function (db) {
         return self.beforeFindListeners.fire(query, req).then(function () {
 
-            return q.ninvoke(db.collection(self.name).find(query), "toArray").then(function(res) {
+            return q.ninvoke(db.collection(self.name).find(query, req), "toArray").then(function(res) {
```

Comment:
Fantastic! Thank you.

Maybe you could use rest call to odata endpoint to get the list of templates.
`/odata/templates?$select=Name`